ui/page/components/utils.go

```diff
@@ -12,3 +19,18 @@ func ContextDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+func RetryFunc(attempts int, sleep time.Duration, errFunc func() error) (att int, err error) {
+	for i := 0; i < attempts; i++ {
+		if i > 0 {
+			log.Println("retrying after error:", err)
+			time.Sleep(sleep)
+			sleep *= 2
```

Comment:
I'd reorder this, this way (PS: for logging purposes, it'd be helpful to pass a `funcDesc string` parameter that can be used in the suggested log below, I also renamed `sleep` to `sleepDur`):
```suggestion
			sleepDur *= i
			log.Println("waiting %s to retry function %s after error: %v", sleepDur, funcDesc, err)
			time.Sleep(sleepDur)
```

Reply:
`sleepDur *= i` `i` does not work... Mismatch type.